tezos/api/src/ocaml_conv/from_ocaml.rs

```diff
@@ -0,0 +1,194 @@
+{
+    fn from_ocaml(v: OCaml<OperationProtocolDataJsonWithErrorListJson>) -> Self {
+        unpack_ocaml_block! { v =>
+            OperationProtocolDataJsonWithErrorListJson {
+                protocol_data_json: OCamlBytes,
+                error_json: OCamlBytes,
+            }
+        }
+    }
+}
+
+unsafe impl FromOCaml<Errored> for Errored {
+    fn from_ocaml(v: OCaml<Errored>) -> Self {
+        unpack_ocaml_block! { v =>
+            Errored {
+                hash: OCamlOperationHash,
+                is_endorsement: Option<bool>,
+                protocol_data_json_with_error_json: OperationProtocolDataJsonWithErrorListJson,
+            }
+        }
+    }
+}
+
+unsafe impl FromOCaml<ValidateOperationResult> for ValidateOperationResult {
+    fn from_ocaml(v: OCaml<ValidateOperationResult>) -> Self {
+        unpack_ocaml_block! { v =>
+            ValidateOperationResult {
+                applied: OCamlList<Applied>,
+                refused: OCamlList<Errored>,
+                branch_refused: OCamlList<Errored>,
+                branch_delayed: OCamlList<Errored>,
+            }
+        }
+    }
+}
+
+unsafe impl FromOCaml<ValidateOperationResponse> for ValidateOperationResponse {
+    fn from_ocaml(v: OCaml<ValidateOperationResponse>) -> Self {
+        unpack_ocaml_block! { v =>
+            ValidateOperationResponse {
+                prevalidator: PrevalidatorWrapper,
+                result: ValidateOperationResult,
+            }
+        }
+    }
+}
+
+unsafe impl FromOCaml<JsonRpcResponse> for JsonRpcResponse {
+    fn from_ocaml(v: OCaml<JsonRpcResponse>) -> Self {
+        unpack_ocaml_block! { v =>
+            JsonRpcResponse {
+                body: OCamlBytes,
+            }
+        }
+    }
+}
+
+// TODO: remove this after TE-207 has been solved
+unsafe impl FromOCaml<Path> for FfiPath {
+    fn from_ocaml(v: OCaml<Path>) -> Self {
+        if v.is_long() {
+            FfiPath(Path::Op)
+        } else {
+            match v.tag_value() {
+                0 => {
+                    let path: FfiPath = unsafe { v.field::<Path>(0).into_rust() };
+                    let right = unsafe { v.field::<OCamlHash>(1).into_rust() };
+
+                    FfiPath(Path::Left(Box::new(PathLeft::new(
+                        path.0,
+                        right,
+                        Default::default(), // TODO: what is body?
+                    ))))
+                }
+                1 => {
+                    let left = unsafe { v.field::<OCamlHash>(0).into_rust() };
+                    let path: FfiPath = unsafe { v.field::<Path>(1).into_rust() };
+
+                    FfiPath(Path::Right(Box::new(PathRight::new(
+                        left,
+                        path.0,
+                        Default::default(), // TODO: what is body?
+                    ))))
+                }
+                tag => panic!("Invalid tag value for OCaml<Path>: {}", tag),
```

Comment:
@tizoc 
I am not sure about this panic, this could cause unexpected behavior, it will probably just crash protocol_runner subprocess, but there is also rust trait `impl TryFrom` which return Result for this cases

Reply:
In theory this conversion will never fail, *unless* tezos changes the representation for its `Path` values or memory gets corrupted somehow (which then would crash things anyway at some other place).

Reply:
ok, ok, I understand,
so please, just add here some comment, that this `tag =>` is reached only if tezos changes internally something,
maybe we will come back here later when we will check clippy or something like that, so that comment would be usefull

Reply:
Ok, will do. At some point I will come up with a good safe abstraction for mapping between Rust<->OCaml structs and enums without the need for so much manually written code, and then this will not be needed anymore.